src/lib.rs

```diff
@@ -3,15 +3,34 @@ extern crate libc;
 #[cfg(target_os = "openbsd")]
 mod openbsd;
 
+use std::ffi::{CString, NulError};
+
 #[derive(Debug, PartialEq)]
 pub enum Error {
     NotSupported,
+    Path(NulError),
+    Permissions(NulError),
     Os(i32),
 }
 
 #[cfg(target_os = "openbsd")]
 pub fn unveil(path: impl AsRef<[u8]>, permissions: &str) -> Result<(), Error> {
-    openbsd::unveil(path, permissions).map_err(Error::Os)
+    let path = path.as_ref();
+
+    // iff path is empty, pass (NULL, NULL) to lock unveil(2). POSIX
```

Comment:
Small typo
```suggestion
    // if path is empty, pass (NULL, NULL) to lock unveil(2). POSIX
```

Reply:
iff was intentional (it means [if and only if](https://en.wikipedia.org/wiki/If_and_only_if)).